src/book.py

```diff
@@ -58,7 +58,9 @@ def set_price(self, price: float) -> None:
         self.__price = price
 
     def __eq__(self, other):
-        return other.__title == self.__title
+        if isinstance(other, Book):
+            return other.__title == self.__title
+        return other in self.__title
```

Comment:
Aqui não deveria retornar false se não é um link e o título é diferente? Você assume que se não é um livro, é uma string e faz uma pesquisa por isso? Acha que é uma boa ideia ter um método que funciona de duas formas diferentes?

Reply:
Concordo que ficou confuso, fiz isso para poder utilizar a palavra reservada **in**, mas entendo que por tornar o código mais complexo não valha a pena.